pandas/core/dtypes/base.py

```diff
@@ -424,20 +424,23 @@ def register(self, dtype: type[ExtensionDtype]) -> None:
 
         self.dtypes.append(dtype)
 
-    def find(self, dtype: type[ExtensionDtype] | str) -> type[ExtensionDtype] | None:
+    def find(
+        self, dtype: type[ExtensionDtype] | ExtensionDtype | str
+    ) -> type[ExtensionDtype] | ExtensionDtype | None:
         """
         Parameters
         ----------
-        dtype : Type[ExtensionDtype] or str
+        dtype : Type[ExtensionDtype] or ExtensionDtype or str
```

Comment:
`Type[ExtensionDtype] or ExtensionDtype` -> `ExtensionDtype class or instance`?

Reply:
Done in latest commit.  Also added overloads to clarify behavior as described here https://github.com/pandas-dev/pandas/pull/40421/files#r622387729 which allows a type-ignore to be removed in `pandas/core/dtypes/common.py`